### PR TITLE
Employee tree

### PR DESCRIPTION
**Pull request by Zlash65**
employee-nested
Without nestedset - [#11621 ](https://github.com/frappe/erpnext/pull/11621)

### CODE REVIEW

**Comment by Olawale1** [comment]
Hi,

Good job. Hope all branches will be expanded by default. Would be odd to open an Employee tree only to find the names of top management and then having to click each name to view subordinates 

Kind regards,

**Comment by Zlash65** [comment]
@Olawale1 No it does not by default as of now but it could be done, I'll ask around and see what majority has to say and will implement it 👍

**Comment by rohitwaghchaure** [comment]
@Zlash65 
Patch is required to update lft and rgt of existing employee

File: erpnext/hr/doctype/employee/employee.py [hunk 1]
```diff
@@ -300,4 +302,26 @@ def get_employee_emails(employee_list):
...
\ No newline at end of file
+	return employee_emails
+
+@frappe.whitelist()
+def get_children(doctype, parent=None, company=None, is_root=False, is_tree=False):
+	condition = ''
+
+	if is_root:
+		parent = ""
+	if parent and company and parent!=company:
```
Comment (rohitwaghchaure) [inline erpnext/hr/doctype/employee/employee.py]: Why checked company with parent?

File: erpnext/hr/doctype/employee/employee.py [hunk 2]
```diff
@@ -52,7 +53,11 @@ def validate(self):
 				frappe.permissions.remove_user_permission(
 					"Employee", self.name, existing_user_id)
 
+	def update_nsm_model(self):
+		frappe.utils.nestedset.update_nsm(self)
+
 	def on_update(self):
+		self.update_nsm_model()
```
Comment (rohitwaghchaure) [inline erpnext/hr/doctype/employee/employee.py]: Add in on_trash too!

**Comment by rohitwaghchaure** [comment]
Closing PR, kindly make changes and reopen it

File: erpnext/hr/doctype/employee/employee.py [hunk 3]
```diff
@@ -154,13 +159,13 @@ def validate_reports_to(self):
 			throw(_("Employee cannot report to himself."))
 
 	def on_trash(self):
+		self.update_nsm_model()
```
Comment (nabinhait) [inline erpnext/hr/doctype/employee/employee.py]: Instead of calling `update_nsm_model`, call super function (on_trash of NestedSet class)

**Review by nabinhait — CHANGES_REQUESTED** [review]

**Review by nabinhait — CHANGES_REQUESTED** [review]
- Renaming of employee breaks
- Merging also breaks
The error comes from `before_rename` and `after_rename` function of nestedset.py

**Pull request merged**